kernel/names.mli

```diff
@@ -499,6 +499,13 @@ val constructor_user_hash : constructor -> int
 val constructor_syntactic_ord : constructor -> constructor -> int
 val constructor_syntactic_hash : constructor -> int
 
+(** {6 Global reference is a kernel side type for all references together } *)
```

Comment:
I'm basically ok with moving global_reference here, which should probably allow more uniform APIs in other parts of the kernel as well.

Reply:
Indeed this change needs wider review @ppedrot @herbelin , I am not expert enough to see if that helps the kernel [which is currently not using any `global_reference`] but it sure helps a bit some other parts.

@mattam82 , what parts would benefit from using `global_reference` ?